### PR TITLE
add cJSON lib

### PR DESCRIPTION
**Pull request by bucanero**
simple JSON library (ANSI C)

### CODE REVIEW

File: cjson/PSPBUILD [hunk 1]
```diff
@@ -0,0 +1,30 @@
+pkgname=cJSON
```
Comment (sharkwouter) [inline cjson/PSPBUILD:1]: Can you change the name to only have lower case letters?

File: cjson/PSPBUILD [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+pkgrel=1
+pkgdesc="Ultralightweight JSON parser in ANSI C."
+arch=('mips')
+url="https://github.com/DaveGamble/cJSON/"
+license=('MIT license')
+depends=()
+makedepends=()
+optdepends=()
+source=("https://github.com/DaveGamble/cJSON/archive/refs/tags/v1.7.16.tar.gz")
+sha256sums=('SKIP')
```
Comment (sharkwouter) [inline cjson/PSPBUILD:12]: Please add a checksum

File: cjson/PSPBUILD [hunk 3]
```diff
@@ -0,0 +1,30 @@
+pkgname=cJSON
+pkgver=1.7.16
+pkgrel=1
+pkgdesc="Ultralightweight JSON parser in ANSI C."
+arch=('mips')
+url="https://github.com/DaveGamble/cJSON/"
+license=('MIT license')
```
Comment (sharkwouter) [inline cjson/PSPBUILD:7]:
```suggestion
license=('MIT')
```

File: cjson/PSPBUILD [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    cd "$pkgname-$pkgver/build"
+    cmake -Wno-dev -DCMAKE_TOOLCHAIN_FILE=$PSPDEV/psp/share/pspdev.cmake -DCMAKE_INSTALL_PREFIX=${pkgdir}/psp -DBUILD_SHARED_LIBS=OFF \
+        -DENABLE_CJSON_UTILS=On -DENABLE_LOCALES=OFF -DENABLE_CJSON_TEST=OFF "${XTRA_OPTS[@]}" .. || { exit 1; }
+    make --quiet $MAKEFLAGS || { exit 1; }
+}
+
+package() {
+    cd "$pkgname-$pkgver/build"
+    make --quiet $MAKEFLAGS install
+    cd ..
```
Comment (sharkwouter) [inline cjson/PSPBUILD:29]:
```suggestion
```

**Comment by sharkwouter** [comment]
Cool, thanks! I made some comments, but it looks generally good and I'd love to have this library

File: cjson/PSPBUILD (hunk 1 above)
Comment (bucanero) [reply cjson/PSPBUILD:1]: oh I used the original casing so I could use `cd "$pkgname-$pkgver` later on the scripts.

I'll change the pkgname but I'll have to hardcode the bash scripts. (source folder will still be `cJSON-version`)

File: cjson/PSPBUILD (hunk 2 above)
Comment (bucanero) [reply cjson/PSPBUILD]: I added the checksum. 👍 
Note: I wasn't sure if Github actually keeps the same file or "regenerates" the .tgz or .zip release file.

Comment (sharkwouter) [reply cjson/PSPBUILD]: Thanks! It should

**Review by sharkwouter — CHANGES_REQUESTED** [review]
Thanks for updating! The PSPBUILD looks much better now. I looked at the package now and I noticed 2 things which need changing, after that this can be merged.

1. The license is still missing from the package. You can add it by adding the following to the package step:
```
    mkdir -m 755 -p "$pkgdir/psp/share/licenses/$pkgname"
    install -m 644 ../LICENSE "$pkgdir/psp/share/licenses/$pkgname"
```
2. The build directory's full path is still referenced in the `.pc` and `.cmake` files that are generated. The `.pc` files should instead be using the `$PSPDEV` environment variable in directory names. CMake can be made to use relative paths or use `$PSPDEV`. There is an example of how this could be done in the sdl2-gfx package for example: https://github.com/pspdev/psp-packages/blob/1e60e62d9cad6782bc22640349b576a7b222bd26/sdl2-gfx/PSPBUILD#L14-L20

You can see if this has been resolved in the output of the build. Right now it looks like this:
```
==> Checking for packaging issues...
==> WARNING: Package contains reference to $pkgdir
psp/lib/cmake/cJSON/cjson.cmake
psp/lib/cmake/cJSON/cJSONConfig.cmake
psp/lib/cmake/cJSON/cjson_utils.cmake
psp/lib/cmake/cJSON/cjson_utils-noconfig.cmake
psp/lib/cmake/cJSON/cjson-noconfig.cmake
psp/lib/pkgconfig/libcjson.pc
psp/lib/pkgconfig/libcjson_utils.pc
```

Sorry if this is a lot. Let me know if you need support, then I'll help you out.

**Comment by sharkwouter** [comment]
Thanks! Looks good now

**Pull request merged**